Pretty css hack to debug layouts (#23)
cybertec-postgresql/today-i-learned-content: posts/TEMPLATE/index.md

---
date: 2030-12-31 # Auto updates on publish.
title: Pretty CSS Hack to debug layouts
author: agustin.ramirez@cybertec.at

> tags: ["css", "hack", "debug-layout"] # max. 10 tags; lowercase; dash-separated

Reviewer: ```suggestion
tags: ["css", "hack", "debug", "layout"] # max. 10 tags; lowercase; dash-separated
```

> description: "Quick screenshots of your layouts to see if everything it's OK" # max. 300 chars.

Reviewer: ```suggestion
description: "Bookmark to temporarily visualize layout of a website" # max. 300 chars.
```

---

> 1. Go to add a new Bookmark in your browser.

Reviewer: ```suggestion
1. Create a new bookmark
```

> 2. Add the name that you want.

Reviewer: ```suggestion
```

> 3. On location/URL field add this code:

Reviewer: ```suggestion
2. Add the following code to the bookmark URL:
```

	var items = [];
	for (var i = 0; i < elements.length; i++) {
		if (elements[i].innerHTML.indexOf('* { background:#000!important;color:#0f0!important;outline:solid #f00 1px!important; background-color: rgba(255,0,0,.2) !important; }') != -1) {
			items.push(elements[i]);
		}
	}
	if (items.length > 0) {
		for (var i = 0; i < items.length; i++) {
			items[i].innerHTML = '';
		}
	} else {
		document.body.innerHTML +=
			'<style>* { background:#000!important;color:#0f0!important;outline:solid #f00 1px!important; background-color: rgba(255,0,0,.2) !important; }\
            * * { background-color: rgba(0,255,0,.2) !important; }\
            * * * { background-color: rgba(0,0,255,.2) !important; }\
            * * * * { background-color: rgba(255,0,255,.2) !important; }\
            * * * * * { background-color: rgba(0,255,255,.2) !important; }\
            * * * * * * { background-color: rgba(255,255,0,.2) !important; }\
            * * * * * * * { background-color: rgba(255,0,0,.2) !important; }\
            * * * * * * * * { background-color: rgba(0,255,0,.2) !important; }\
            * * * * * * * * * { background-color: rgba(0,0,255,.2) !important; }</style>';
	}
})();
```

> 4. Save.

Reviewer: ```suggestion
```

> To see how it's works, you just to click on the bookmark name that you defined, and see you the magic!

Reviewer: ```suggestion
To use it, just navigate to a website and click on the bookmark you defined.
The image below shows [this website](https://www.cybertec-postgresql.com/en/) with the bookmark activated.
```

![Screenshot of Cybertec Layout](./screenshot.png)

You can use it on any page.

> Is it cool no? :)

Reviewer: ```suggestion
Ain't that cool? 😀
```

> P.s: Tested on Chrome and Firefox.

Reviewer 2: ```suggestion
P.S.: Tested on Chrome and Firefox.
```
